components/TradingView/index.tsx

```diff
@@ -123,29 +196,125 @@ const TVChartContainer = () => {
+                2
+              )}) is greater than 5% ${
+                order.side == 'buy' ? 'above' : 'below'
+              } the current market price ($${markPrice.toFixed(2)}). ` +
+              ' indicating you might incur significant slippage. <p><p>Please use the trade input form if you wish to accept the potential slippage.',
+            callback: () => {
+              this.setPrice(currentOrderPrice)
+            },
+          })
+          selectedMarketPrice
+        } else {
+          tvWidgetRef.current.showConfirmDialog({
+            title: 'Change Order Price?',
+            body: `Would you like to change your order from a 
+           ${order.size} ${order.marketName.split('/')[0].toUpperCase()} ${
+              order.side
+            } at $${currentOrderPrice} 
+           to a 
+          ${order.size} ${order.marketName.split('/')[0].toUpperCase()} ${
+              order.side
+            } at $${updatedOrderPrice} 
+          `,
+            callback: (res) => {
+              if (res) {
+                handleChangeOrder(order, updatedOrderPrice).then((result) => {
+                  if (result) {
+                    lines.delete(order.orderId.toString())
+                    this.remove()
+                  } else {
+                    this.setPrice(currentOrderPrice)
+                  }
+                })
+              } else {
+                this.setPrice(currentOrderPrice)
+              }
+            },
+          })
+        }
+      })
+      .onCancel(function () {
+        tvWidgetRef.current.showConfirmDialog({
+          title: 'Cancel Your Order?',
+          body: `Would you like to cancel your order for 
+       ${order.size} ${order.marketName.split('/')[0].toUpperCase()} ${
+            order.side
+          } at $${order.price}  
+      `,
+          callback: (res) => {
+            if (res) {
+              handleCancelOrder(order).then((res) => {
+                if (res) {
+                  lines.delete(order.orderId.toString())
+                  this.remove()
+                }
+              })
+            }
+          },
+        })
+      })
+      .setText(
+        `${order.side.toUpperCase()} ${order.marketName
+          .split('/')[0]
+          .toUpperCase()}`
+      )
+      .setBodyBorderColor(order.side == 'buy' ? '#AFD803' : '#E54033')
+      .setBodyBackgroundColor('#000000')
+      .setBodyTextColor('#F2C94C')
+      .setLineLength(3)
+      .setLineColor(order.side == 'buy' ? '#AFD803' : '#E54033')
+      .setQuantity(order.size)
+      .setTooltip(`Order #: ${order.orderId}`)
+      .setQuantityBorderColor(order.side == 'buy' ? '#AFD803' : '#E54033')
+      .setQuantityBackgroundColor('#000000')
+      .setQuantityTextColor('#F2C94C')
+      .setCancelButtonBorderColor(order.side == 'buy' ? '#AFD803' : '#E54033')
+      .setCancelButtonBackgroundColor('#000000')
+      .setCancelButtonIconColor('#F2C94C')
```

Comment:
@saml33 can you propose how we can integrate this with the themes?

Reply:
Yep, I think we can do it with this https://tailwindcss.com/docs/configuration#referencing-in-java-script